### PR TITLE
[CWS] add process event monitor and event subscriber interface

### PR DESCRIPTION
**Pull request by safchain**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]
Maybe rebase before merge (for the `go.sum` change) but LGTM

File: pkg/security/probe/probe.go [hunk 1]
```diff
@@ -309,17 +309,23 @@ func (p *Probe) Start() {
...
-	if p.handler != nil {
-		p.handler.HandleEvent(event)
+	// send wildcard first
+	for _, handler := range p.handlers[model.UnknownEventType] {
+		handler.HandleEvent(event)
+	}
+
+	// send specific event
+	for _, handler := range p.handlers[event.GetEventType()] {
+		handler.HandleEvent(event)
```
Comment (mbotarro) [inline pkg/security/probe/probe.go:328]: Do we need to send events to custom handlers if `AgentMonitoringEvents` is not enabled? I wonder if we could add this check here as well
```
if p.config.AgentMonitoringEvents {
}
```

**Review by mbotarro — APPROVED** [review]
Thanks a lot for adding the `ProcessMonitoring` handler! 🙇

File: pkg/security/probe/probe.go (hunk 1 above)
Comment (safchain) [reply pkg/security/probe/probe.go:328]: I think custom handlers are only added if enabled https://github.com/DataDog/datadog-agent/pull/11776/files#diff-3d1a943843e44b06eb76fd0d68d08665d3dafbfa6c9f5d48f928c17e2b5771c8R115 ?

**Review by djmitche — APPROVED** [review]
pkg/config/config.go 👍🏽

File: pkg/security/probe/probe.go (hunk 1 above)
Comment (safchain) [reply pkg/security/probe/probe.go:328]: `AgentMonitoringEvents` config parameter if used by the agent to send internal health custom events

Comment (mbotarro) [reply pkg/security/probe/probe.go:328]: Ohh, that makes sense. Thanks for clarifying it!!

**Review by lebauce — APPROVED** [review]

**Pull request merged**